internal/entities/pokemon.go

```diff
@@ -29,25 +30,37 @@ func BuildPokemonProductFromModel(model models.PokemonModel) *PokemonProduct {
 	}
 
 	tier := Tier{
-		Id: model.Tier.Id,
-		Name: model.Tier.Name,
+		Id:                model.Tier.Id,
+		Name:              model.Tier.Name,
 		MinimalExperience: model.Tier.MinimalExperience,
-		LimitExperience: model.Tier.LimitExperience,
+		LimitExperience:   model.Tier.LimitExperience,
 	}
 
 	return &PokemonProduct{
 		Pokemon: Pokemon{
-			Id: model.Id,
+			Id:          model.Id,
 			ReferenceId: model.ReferenceId,
-			Name: model.Name,
-			Weight: model.Weight,
-			Height: model.Height,
-			ImageUrl: model.ImageUrl,
-			Experience: model.Experience,
-			Types: types,
-			Tier: tier,
+			Name:        model.Name,
+			Weight:      model.Weight,
+			Height:      model.Height,
+			ImageUrl:    model.ImageUrl,
+			Experience:  model.Experience,
+			Types:       types,
+			Tier:        tier,
 		},
-		Price: model.Prices[0].Value,
+		Price:   model.Prices[0].Value,
 		InStock: model.Stock.Quantity,
 	}
 }
+
+func BuildManyPokemonProductFromModel(data []models.PokemonModel) []PokemonProduct {
+	var result []PokemonProduct
+
+	for _, val := range data {
+		var tmp *PokemonProduct
+		tmp = BuildPokemonProductFromModel(val)
+		result = append(result, *tmp)
+	}
```

Comment:
Aqui dava para declarar e atribuir direto.
```suggestion
	for _, val := range data {
		tmp := BuildPokemonProductFromModel(val)
		result = append(result, *tmp)
	}
```